api/src/glfs-fops.c

```diff
@@ -1801,8 +1808,21 @@ glfs_pwritev_common(struct glfs_fd *glfd, const struct iovec *iovec, int iovcnt,
     if (ret)
         gf_msg_debug("gfapi", 0, "Getting leaseid from thread failed");
 
-    ret = syncop_writev(subvol, fd, &iov, 1, offset, iobref, flags, &preiatt,
-                        &postiatt, fop_attr, NULL);
+    if (prestat) {
+        if (poststat)
+            ret = syncop_writev(subvol, fd, &iov, 1, offset, iobref, flags,
+                                &preiatt, &postiatt, fop_attr, NULL);
+        else
+            ret = syncop_writev(subvol, fd, &iov, 1, offset, iobref, flags,
+                                &preiatt, NULL, fop_attr, NULL);
+    } else {
+        if (poststat)
+            ret = syncop_writev(subvol, fd, &iov, 1, offset, iobref, flags,
+                                NULL, &postiatt, fop_attr, NULL);
+        else
+            ret = syncop_writev(subvol, fd, &iov, 1, offset, iobref, flags,
+                                NULL, NULL, fop_attr, NULL);
+    }
```

Comment:
I think this adds too much noise just to save a small memory copy.

Reply:
ACK - reverted.